bluebottle/tasks/models.py

```diff
@@ -181,22 +186,20 @@ def deadline_to_apply_reached(self):
                 "The deadline to apply for your task '{0}' has passed"
             ).format(self.title)
 
-        send_deadline_to_apply_passed_mail(self, subject, connection.tenant)
-
         if self.status == self.TaskStatuses.open:
             if self.people_applied:
                 if self.people_applied + self.externals_applied < self.people_needed:
                     self.people_needed = self.people_applied
-
                 if self.type == self.TaskTypes.ongoing:
-                    self.set_in_progress()
+                    self.status = self.TaskStatuses.in_progress
                 else:
-                    self.set_full()
```

Comment:
This pattern is used one more time, also calling save directly and redundantly after the set. Maybe remove it everywhere?